ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -242,6 +242,7 @@ private void scheduleCompactionIfRequired(CompactionInfo ci, Table t, Partition
     try {
       ValidWriteIdList validWriteIds = resolveValidWriteIds(t);
       CompactionType type = checkForCompaction(ci, validWriteIds, sd, t.getParameters(), runAs);
+      ci.type = type;
```

Comment:
I don't think this is necessary as we are passing compaction type in the request method. Type is set on CompactionRequest object.

Reply:
This required to log the proper type in all cases. In case of changing the type, and later failing, without this change, the original type would be logged.